server/autolinkplugin/plugin.go

```diff
@@ -61,7 +62,26 @@ func contains(team string, channel string, list []string) bool {
 	return false
 }
 
+func (p *Plugin) isBotUser(userId string) bool {
+	user, appErr := p.API.GetUser(userId)
+	if appErr != nil {
+		// NOTE: Not sure how we want to handle errors here, we can either:
+		// * assume that occasional rewrites of Bot messges are ok
+		// * assume that occasional not rewriting of all messages is ok
+		// Let's assume for now that former is a lesser evil
+		mlog.Error("failed to check if message for rewriting was send by a bot", zap.Error(appErr))
+		return false
```

Comment:
Why are you logging the error here instead of returning it?

Reply:
According to [1], the only option we have while handling Post message in a hook is to either reject it, or accept it as is. Even if I return the error to the caller, there is not much we can do with it. So the idea is to simply procces posts which for which we cannot establish origin (bot/non-bot). 

Happy to discuss.


[1] https://github.com/mattermost/mattermost-server/blob/57f3ca975f565b0235d72ba6eb098e73a25513cd/plugin/hooks.go#L92-L115

Reply:
My comment was more about the code flow. In go, it's usually up to the caller to decide who an error should be handled (returned, logged, ignored). Hence, my proposal would be to change the signature to ` isBotUser(userId string) (bool, *model.AppError)` and move the logic to `ProcessPost`.

Reply:
Makes sense. Done.